src/platform/input/sdl2mouse.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief SDL2 Mouse Input interface.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "sdl2mouse.h"
+#include "editmain.h"
+#include "gameclient.h"
+#include <cstdio>
+
+SDL_Cursor *SDL2Mouse::s_loadedCursors[CURSOR_COUNT][8];
+
+SDL2Mouse::SDL2Mouse() : m_nextFreeIndex(0), m_nextGetIndex(0) {}
+
+void SDL2Mouse::Init()
+{
+    Mouse::Init();
+
+    std::memset(m_eventBuffer, 0, sizeof(m_eventBuffer));
+    std::memset(s_loadedCursors, 0, sizeof(s_loadedCursors));
+}
+
+void SDL2Mouse::Load_Cursor(const char *file_location, uint32_t load_index, uint32_t load_direction) const
+{
+    if (s_loadedCursors[load_index][load_direction] != NULL) {
+        // Cursor at this position already loaded. Ignoring subsequent load.
+        return;
+    }
+    // TODO: this should parse the ANI file and load the cursor correctly
```

Comment:
What is this comment for? Did you test this? Does it work?

Reply:
It does not work. ANI files need to be parsed and loaded into multiple bitmaps, which are switched dynamically. Animated cursors only exist on windows, so these cursor needs to be set for each frame individually.

Reply:
Ok. So this is something that will be done in a follow up change?

Reply:
Yes. It's a bit hard to program things like these completly blind

Reply:
Hmm. This should be testable on Windows with the game dat + thyme dll. Do you have a Windows install?

Reply:
No. I tried once but didn't really workout all that great

Reply:
If you can install Windows on your machine, then we should be able to get Thyme running in DLL mode. This way you can test all changes.